lib/core/mock/identity-user.service.mock.ts

```diff
@@ -15,153 +15,251 @@
-export let  mockIdentityUser5: IdentityUserModel = { id: 'mock-user-id-5', username: 'userName5', firstName: 'first-name-5', lastName: 'last-name-5', email: 'abcde@xyz.com' };
-
-export const mockIdentityUsers: IdentityUserModel[] = [
-    mockIdentityUser1,
-    mockIdentityUser2,
-    mockIdentityUser3,
-    mockIdentityUser4,
-    mockIdentityUser5
-];
-
-export const mockIdentityRole  = new IdentityRoleModel({ id: 'id-1', name: 'MOCK-ADMIN-ROLE'});
-
-export const mockAvailableRoles = [
-    new IdentityRoleModel({ id: 'mock-role-id-1', name: 'MOCK-ADMIN-ROLE'}),
-    new IdentityRoleModel({ id: 'mock-role-id-2', name: 'MOCK-USER-ROLE'}),
-    new IdentityRoleModel({ id: 'mock-role-id-3', name: 'MOCK_MODELER-ROLE' }),
-    new IdentityRoleModel({ id: 'mock-role-id-5', name: 'MOCK-ROLE-2'})
-];
-
-export const mockAssignedRoles = [
-    new IdentityRoleModel({ id: 'mock-role-id-1', name: 'MOCK-ADMIN-ROLE'}),
-    new IdentityRoleModel({ id: 'mock-role-id-2', name: 'MOCK_MODELER-ROLE' }),
-    new IdentityRoleModel({ id: 'mock-role-id-3', name: 'MOCK-ROLE-1' })
-];
-
-export const mockEffectiveRoles = [
-    new IdentityRoleModel({id: 'mock-role-id-1', name: 'MOCK-ACTIVE-ADMIN-ROLE'}),
-    new IdentityRoleModel({id: 'mock-role-id-2', name: 'MOCK-ACTIVE-USER-ROLE'}),
-    new IdentityRoleModel({id: 'mock-role-id-3', name: 'MOCK-ROLE-1'})
-];
-
-export const mockJoinGroupRequest: IdentityJoinGroupRequestModel = {userId: 'mock-hser-id', groupId: 'mock-group-id', realm: 'mock-realm-name'};
-
-export const mockGroup1 = <IdentityGroupModel> {
-    id: 'mock-group-id-1', name: 'Mock Group 1', path: '/mock', subGroups: []
-};
-
-export const mockGroup2 = <IdentityGroupModel> {
-    id: 'mock-group-id-2', name: 'Mock Group 2', path: '', subGroups: []
-};
-
-export const mockGroups = [
-    <IdentityGroupModel> { id: 'mock-group-id-1', name: 'Mock Group 1', path: '/mock', subGroups: [] },
-    <IdentityGroupModel> { id: 'mock-group-id-2', name: 'Mock Group 2', path: '', subGroups: [] }
-];
-
-export const queryUsersMockApi: any = {
-    oauth2Auth: {
-        callCustomApi: () => {
-            return Promise.resolve(mockIdentityUsers);
-        }
+import { Injectable } from '@angular/core';
+import {
+    IdentityGroupModel,
+    IdentityRoleModel,
+    IdentityUserModel,
+    mockIdentityGroups,
+    IdentityJoinGroupRequestModel,
+    IdentityUserServiceInterface,
+    IdentityUserPasswordModel,
+    IdentityUserQueryCloudRequestModel,
+    IdentityUserQueryResponse
+} from '@alfresco/adf-core';
+import { Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { mockAssignedRoles, mockAvailableRoles, mockEffectiveRoles, mockIdentityUser1, mockIdentityUsers } from './identity-user.mock';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class IdentityUserServiceMock implements IdentityUserServiceInterface {
```

Comment:
It would be nice to call MockService instead of ServiceMock
```suggestion
export class IdentityUserMockService implements IdentityUserServiceInterface {
```

Reply:
I stick to the convention, here is example:
https://github.com/Alfresco/alfresco-ng2-components/blob/develop/lib/core/mock/alfresco-api.service.mock.ts#L25